Include/Rocket/Core/StringBase.inl

```diff
@@ -26,42 +26,42 @@
  */
 
 template< typename T >
-StringBase< T >::StringBase() : value((T*)local_buffer), buffer_size(LOCAL_BUFFER_SIZE), length(0), hash(0)
+StringBase< T >::StringBase() : value(LocalBuffer()), buffer_size(LOCAL_BUFFER_SIZE), length(0), hash(0)
 {
-	value[0] = 0;
+	value[0] = T();
```

Comment:
It's true it's not necessarily null but it is for all *practical* use cases. Or are you doing something more complicated?
It will be for null for all types I anticipated you using for this class, like char, wchar_t etc. If you're using something more complex well then 0 isn't going to work anyway is it? So what's your point?

I didn't see value enough to take it a step further and rely on something more complicated or a newer standard feature like uniform initialization it was just meant to reduces compiler warnings but if you don't like that, fine won't argue.